library/cpp/include/wavemap/core/utils/query/query_accelerator.h

```diff
@@ -104,22 +106,41 @@ class QueryAccelerator<NdtreeBlockHash<CellDataT, dim>> {
   std::array<NodeType*, morton::kMaxTreeHeight<dim>> node_stack{};
 };
 
-// Query accelerator for hashed wavelet octrees
+/**
+ * A class that accelerates queries by caching block and parent node addresses
+ * to speed up data structure traversals, and intermediate wavelet decompression
+ * results to reduce redundant computation.
+ * @note This class is safe to use in a multi-threaded environment. However,
+ *       concurrent calls to a single instance from multiple threads are not.
+ *       Since the accelerator is lightweight and cheap to construct, we
+ *       recommend using a separate instance per thread for the best performance
+ *       and simplicity.
+ */
 template <>
 class QueryAccelerator<HashedWaveletOctree> {
  public:
   static constexpr int kDim = HashedWaveletOctree::kDim;
 
   explicit QueryAccelerator(const HashedWaveletOctree& map) : map_(map) {}
 
+  //! Reset the cache
+  //! @note This method must be called whenever the map changes, not only to
```

Comment:
Is there a way to support an auto-resetting behavior?

Reply:
It's possible, but it would require quite a bit of bookkeeping. The cleanest method I can think of is to look at the `last_updated_timestamp_` we already store for each block in the map. However, this is fragile because our current interfaces also expose mutable pointers/references to map blocks and nodes, so they can be modified without the map class's knowledge. (It would be very painful to recursively evaluate/update the map otherwise, e.g. in our integrators and planners.)
In general, I like the approach we took in the Python interface, where the query accelerator is not exposed to users but automatically gets used internally instead. We could consider adding similar batched accessors in C++ as well, although I think there are still a lot of use cases where users would need direct access to the accelerator (e.g. in search based planners). Lets discuss it sometime in person? :)